awx_collection/plugins/modules/tower_workflow_launch.py

```diff
@@ -61,106 +86,130 @@
 
 
 EXAMPLES = '''
-- name: Launch a workflow
+- name: Launch a workflow with a timeout of 10 seconds
   tower_workflow_launch:
-    name: "Test Workflow"
-  delegate_to: localhost
-  run_once: true
-  register: workflow_results
+    workflow_template: "Test Workflow"
+    timeout: 10
 
-- name: Launch a Workflow with parameters without waiting
+- name: Launch a Workflow with extra_vars without waiting
   tower_workflow_launch:
     workflow_template: "Test workflow"
-    extra_vars: "---\nmy: var"
+    extra_vars:
+      var1: My First Variable
+      var2: My Second Variable
     wait: False
-  delegate_to: localhost
-  run_once: true
-  register: workflow_task_info
 '''
 
-from ..module_utils.ansible_tower import TowerModule, tower_auth_config
-
-try:
-    import tower_cli
-    from tower_cli.api import client
-    from tower_cli.conf import settings
-    from tower_cli.exceptions import ServerError, ConnectionError, BadRequest, TowerCLIError
-except ImportError:
-    pass
+from ..module_utils.tower_api import TowerModule
+import json
+import time
 
 
 def main():
+    # Any additional arguments that are not fields of the item can be added here
     argument_spec = dict(
-        workflow_template=dict(required=True),
-        extra_vars=dict(),
-        wait=dict(default=True, type='bool'),
-        timeout=dict(default=None, type='int'),
+        name=dict(required=True, aliases=['workflow_template']),
+        organization=dict(),
+        inventory=dict(),
+        limit=dict(),
+        scm_branch=dict(),
+        extra_vars=dict(type='dict'),
+        wait=dict(required=False, default=True, type='bool'),
+        interval=dict(required=False, default=1.0, type='float'),
+        timeout=dict(required=False, default=None, type='int'),
     )
 
-    module = TowerModule(
-        argument_spec=argument_spec,
-        supports_check_mode=True
-    )
+    # Create a module for ourselves
+    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
-    workflow_template = module.params.get('workflow_template')
-    extra_vars = module.params.get('extra_vars')
+    optional_args = {}
+    # Extract our parameters
+    name = module.params.get('name')
+    organization = module.params.get('organization')
+    inventory = module.params.get('inventory')
+    optional_args['limit'] = module.params.get('limit')
+    optional_args['extra_vars'] = module.params.get('extra_vars')
```

Comment:
This line should be deleted because `optional_args['extra_vars']` gets overwritten on line 139

Reply:
Oops!  Deleted.